backend/__init__.py

```diff
@@ -0,0 +1,142 @@
+import json
+
+from elasticsearch import Elasticsearch
+from flask import Flask, g, jsonify
+from flask_cors import CORS
+from flask_login import LoginManager
+from flask_mongoengine import MongoEngine
+from flask_security import Security, MongoEngineUserDatastore, UserMixin, RoleMixin
+from flask_security.utils import encrypt_password
+from jsonschema import FormatChecker, Draft4Validator
+from pymongo import MongoClient, ReturnDocument
+from mongoengine.fields import UUIDField, ListField, StringField, BooleanField
+from werkzeug.routing import BaseConverter
+
+from api.projects import projects
+from api.users import users
+from api.search import search
+from api.helper.apiexception import ApiException
+
+app = Flask(__name__, static_url_path='')
+CORS(app)
+
+app.config['DEBUG'] = True
+app.config['SECRET_KEY'] = 'super-secret'
+app.config['MONGODB_DB'] = 'knexdb'
+app.config['MONGODB_HOST'] = 'mongodb'
+app.config['MONGODB_PORT'] = 27017
+app.config['SECURITY_PASSWORD_HASH'] = 'pbkdf2_sha512'
+app.config['SECURITY_PASSWORD_SALT'] = 'THISISMYOWNSALT'
+app.config['UPLOAD_FOLDER'] = ''
+app.config['MAX_CONTENT_PATH'] = 1000000  # 100.000 byte = 100kb
+
+global DB
+DB = MongoEngine(app)
+
+@app.before_first_request
+def init_global_elasticsearch():
+    global ES
+    ES = Elasticsearch([{'host': 'elasticsearch', 'port': 9200}])
+
+@app.before_request
+def set_global_elasticsearch():
+    g.es = ES
+
+@app.before_first_request
+def init_gloabl_mongoclient():
+    global MONGOCLIENT
+    MONGOCLIENT = MongoClient('mongodb:27017')
+
+@app.before_request
+def set_global_mongoclient():
+    g.knexdb = MONGOCLIENT.knexdb
+    g.projects = g.knexdb.projects
+
+@app.before_first_request
+def init_global_login_manager():
+    global LOGINMANAGER
+    LOGINMANAGER = LoginManager()
+    LOGINMANAGER.init_app(app)
+
+@app.before_request
+def init_global_manifest_validator():
+    with app.open_resource("manifest_schema.json", mode='r') as schema_file:
+        schema = json.load(schema_file)
+        global VALIDATOR
+        VALIDATOR = Draft4Validator(schema, format_checker=FormatChecker())
```

Comment:
this doesn't have to be called every request since it's a global, not member of flasks "g" object